### DISCUSSION TITLE
Possible problem in example "Distance between two surfaces"

### DISCUSSION THREAD

**Discussion by scarpma** [pyvista/pyvista#2527] [Q&A] [answered]
Hello,

I have some doubts about these two lines:
https://github.com/pyvista/pyvista/blob/9a1373dabf314b0142470c757207c9e9ff1a7cf3/examples/01-filter/distance-between-surfaces.py#L75-L76

I think that giving `p0` as `p-vec` will result in a wrong distance since the measure is not starting from the surface, but from points of the surface with an offset of `-vec`.

**Comment by MatthewFlamm** [comment, marked as answer]
Here is the rest of that section:

https://github.com/pyvista/pyvista/blob/9a1373dabf314b0142470c757207c9e9ff1a7cf3/examples/01-filter/distance-between-surfaces.py#L73-L79

The `ray_trace` method returns the position of the point in the surface that is intersected by the ray.  That is, the distance from p0 isn't important.  Further down you can see that the distance to this point, `ip` is computed from `p` and not `p0`.

**Comment by scarpma** [comment]
Ahh ok perfect. It makes perfect sense. Thanks for the clarification.

**Comment by MatthewFlamm** [comment]
I'm going to move this to a discussion so it can stay "open" for others to find.
